extension/src/background/index.ts

```diff
@@ -8,17 +8,40 @@ async function polling() {
     log(`Polling`);
 }
 
+async function fetchPreviousRoomId() {
+    return new Promise(function (resolve, reject) {
+        // Maybe boot a random roomId on first load ?
+        chrome.storage.sync.get({ roomId: undefined }, function (options) {
+            resolve(options.roomId);
+        });
+    });
+}
+// TODO : Connect to previous roomId using websocket
+
 Poller.getInstance(polling);
 
 var portFromCS: Runtime.Port;
 
 const connected = (p: Runtime.Port) => {
     portFromCS = p;
-    portFromCS.postMessage("Sending message to CS");
+    console.log("[BG] Content Script connected");
+    portFromCS.postMessage("[BG] Sending message to CS");
+
     portFromCS.onMessage.addListener((m: any) => {
-        console.log("[BG] Message from CS");
-        console.log(m);
+        console.log(`[BG] Received message ${m} from content script`);
+        switch (m.type) {
```

Comment:
I agree with the pattern. We'll need to merge it with what I proposed in my PR on video events.
We should use typescript to help us here (not use any) and define all the type that can be used.

Using this pattern, we could have a different "body" for different type and still keep typescript type to help us know what the properties the object has

Reply:
I have tried to look at how to define custom typescript types for postMessage, but I could not find anything / I'm not knowledgeable enough in typescript. Any help appreciated